docs/source/model_doc/yoso.mdx

```diff
@@ -0,0 +1,80 @@
+<!--Copyright 2022 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# YOSO
+
+## Overview
+
+The YOSO model was proposed in [You Only Sample (Almost) Once: Linear Cost Self-Attention Via Bernoulli Sampling](https://arxiv.org/abs/2111.09714)  
+by Zhanpeng Zeng, Yunyang Xiong, Sathya N. Ravi, Shailesh Acharya, Glenn Fung, Vikas Singh. 
+
+The abstract from the paper is the following:
+
+*Transformer-based models are widely used in natural language processing (NLP). Central to the transformer model is 
+the self-attention mechanism, which captures the interactions of token pairs in the input sequences and depends quadratically 
+on the sequence length. Training such models on longer sequences is expensive. In this paper, we show that a Bernoulli sampling 
+attention mechanism based on Locality Sensitive Hashing (LSH), decreases the quadratic complexity of such models to linear. 
+We bypass the quadratic cost by considering self-attention as a sum of individual tokens associated with Bernoulli random 
+variables that can, in principle, be sampled at once by a single hash (although in practice, this number may be a small constant). 
+This leads to an efficient sampling scheme to estimate self-attention which relies on specific modifications of 
+LSH (to enable deployment on GPU architectures). We evaluate our algorithm on the GLUE benchmark with standard 512 sequence 
+length where we see favorable performance relative to a standard pretrained Transformer. On the Long Range Arena (LRA) benchmark, 
+for evaluating performance on long sequences, our method achieves results consistent with softmax self-attention but with sizable 
+speed-ups and memory savings and often outperforms other efficient self-attention methods. Our code is available at this https URL*
+
+This model was contributed by [novice03](https://huggingface.co/novice03)`. The original code can be found [here](https://github.com/mlpen/YOSO).
```

Comment:
We generally add tips here regarding the model: 
<img width="774" alt="image" src="https://user-images.githubusercontent.com/30755778/150150300-d88a7e99-f3d0-4aaf-b240-8ea5f3604183.png">

Would it be possible to do the same, and to explain what the custom kernels do?

Reply:
Yea, I added some information about the kernels. Please let me know if you think I should add more details.

Reply:
That looks good, thanks @novice03! I wonder if "correct PyTorch and cudatoolkit versions" could be more explicit or if the error would tell the user which versions should be used?

Reply:
Hello @LysandreJik. The errors do provide some details:

For example, if Ninja isn't installed, there is a RuntimeError in the stacktrace - `RuntimeError: Ninja is required to load C++ extensions`.  

As another test, I tried to use the kernels with the incorrect PyTorch version on a A100 GPU, and I see the following warning: `NVIDIA A100-PCIE-40GB with CUDA capability sm_80 is not compatible with the current PyTorch installation. The current PyTorch install supports CUDA capabilities sm_37 sm_50 sm_60 sm_61 sm_70 sm_75 compute_37.
If you want to use the NVIDIA A100-PCIE-40GB GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/`.

So, I think this should be enough to inform the users what to do if the compilation isn't successful. 

Btw, I moved the function call to load kernels from `YosoSelfAttention` to `YosoEncoder`, because it seems unnecessary to compile the kernels every time a self-attention class is initiated. Instead, it seems more efficient to load it only once - in the encoder.